aquarius/events/processors.py

```diff
@@ -352,3 +381,52 @@ def process(self):
         self.es_instance.update(self.asset, self.did)
 
         return self.asset
+
+
+class MetadataStateProcessor(EventProcessor):
+    def process(self):
+        self.did = make_did(self.event.address, self._chain_id)
+
+        if self.event.args.state != MetadataStates.ACTIVE:
+            try:
+                self.asset = self._es_instance.read(self.did)
+            except Exception:
+                self.asset = None
+            if self.asset and self.asset.get("id") == self.did:
+                self.soft_delete_ddo(self.did)
+            return True
+
+        decryptor_url, *_ = self.dt_contract.caller.getMetaData()
+
+        soft_deleted_ddo = self._es_instance.read(self.did)
+
+        # Reacreate an event with the required params from the data stored of
+        # MetadataCreate/MetadataUpdate events after soft deletion
+        event = AttributeDict(
```

Comment:
why are we manually recreating the event? Can't we retrieve it from the chain based on its event txid?